tests/common/widgets/search-box/options.ts

```diff
@@ -0,0 +1,461 @@
+      }
+    });
+
+    test('shows a reset button when there is a query', async () => {
+      const searchClient = createSearchClient({});
+
+      await setup({
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient,
+        },
+        widgetParams: {},
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      expect(document.querySelector('.ais-SearchBox-reset')).not.toBeVisible();
+
+      await act(async () => {
+        userEvent.type(screen.getByRole('searchbox'), 'iPhone');
+
+        await wait(0);
+      });
+
+      expect(
+        screen.getByRole('button', {
+          name: /clear the search query/i,
+        })
+      ).toBeVisible();
+    });
+
+    test('shows a loader when search is stalled', async () => {
+      let nbSearches = 0;
+      const searchClient = createSearchClient({
+        search: async () => {
+          if (++nbSearches === 2) {
+            await new Promise((resolve) => setTimeout(resolve, 50));
+          }
+
+          return createMultiSearchResponse();
+        },
+      });
+
+      await setup({
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient,
+          stalledSearchDelay: 0,
+        },
+        widgetParams: {},
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      expect(
+        document.querySelector('.ais-SearchBox-loadingIndicator')
+      ).not.toBeVisible();
+
+      await act(async () => {
+        userEvent.type(screen.getByRole('searchbox'), 'a');
+
+        await wait(0);
+      });
+
+      expect(
+        document.querySelector('.ais-SearchBox-loadingIndicator')
+      ).toBeVisible();
+    });
+
+    test('forwards `placeholder` prop', async () => {
+      const searchClient = createSearchClient({});
+
+      await setup({
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient,
+        },
+        widgetParams: { placeholder: 'Search here' },
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      expect(screen.getByRole('searchbox')).toHaveAttribute(
+        'placeholder',
+        'Search here'
+      );
+    });
+
+    test('refines query when typing by default', async () => {
+      let state: UiState = {};
+      const searchClient = createSearchClient({});
+
+      await setup({
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient,
+          onStateChange: ({ uiState }) => (state = uiState),
+        },
+        widgetParams: {},
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      userEvent.type(screen.getByRole('searchbox'), 'iPhone');
+
+      expect(state.indexName.query).toBe('iPhone');
+      expect(screen.getByRole('searchbox')).toHaveValue('iPhone');
+    });
+
+    test('refines when hitting the Enter key', async () => {
```

Comment:
this test also passes when searchAsYouType: false isn't implemented. It should maybe be in a describe so that can be tested and skipped in vue